crates/cairo-coverage/src/input/test_function_filter.rs

```diff
@@ -0,0 +1,27 @@
+use crate::types::FunctionName;
+use cairo_lang_sierra::ids::FunctionId;
+
+pub struct TestFunctionFilter {
+    include_test_functions: bool,
+    test_functions: Vec<String>,
+}
+
+impl TestFunctionFilter {
+    pub fn new(
+        functions_marked_with_test_attribute: Option<&Vec<FunctionId>>,
+        include_test_functions: bool,
+    ) -> Self {
+        Self {
+            include_test_functions,
+            test_functions: functions_marked_with_test_attribute
+                .unwrap_or(&Vec::new())
+                .iter()
+                .map(ToString::to_string)
+                .collect(),
+        }
+    }
+
+    pub fn should_include(&self, function_name: &FunctionName) -> bool {
+        self.include_test_functions || !self.test_functions.contains(function_name)
```

Comment:
What's the rationale here?

Reply:
If the user wants to include the test function, we should always return `true`. Otherwise, we need to check if the function name is not the same as the test function's name, as this would imply that it is a test function, and we don't want this.